Gems/Atom/RPI/Code/Source/RPI.Public/Pass/RenderPass.cpp

```diff
@@ -193,9 +193,11 @@ namespace AZ
         void RenderPass::FrameBeginInternal(FramePrepareParams params)
         {
             m_timestampResult = AZ::RPI::TimestampResult();
+
+            // the pass is potentially re-allocated amount devices dynamically in runtime so deviceIndex is updated everyframe.
             if (GetScopeId().IsEmpty())
             {
-                InitScope(RHI::ScopeId(GetPathName()), m_hardwareQueueClass, m_deviceIndex);
+                InitScope(RHI::ScopeId(GetPathName()), m_hardwareQueueClass, Pass::GetDeviceIndex());
```

Comment:
Nitpick: No need for the `Pass::` namespace.

Reply:
And if I am doing it here, the m_deviceIndex in Pass would have a higher priority then the one in ScopeProducer, is it what it is supposed to?

Reply:
Ah right. Yes, leave it as is.